src/components/Admin/AdminPreferences.tsx

```diff
@@ -19,131 +18,53 @@ export const AdminPreferences: React.FC = () => {
     title: t('PREFERENCES_SAVE_WARNING'),
     message: t('PREFERENCES_SAVE_MESSAGE'),
     confirmText: t('BUTTON_CONFIRM'),
+    awaitAction: false,
   })
 
   const [prefs, setPrefs] = useState<object>()
-  const [hasChanged, setHasChanged] = useState(false)
   const [isSaving, setIsSaving] = useState(false)
-  const [undoQueue, setUndoQueue] = useState<object[]>([])
-  const [redoQueue, setRedoQueue] = useState<object[]>([])
 
   useEffect(() => {
-    getRequest(getServerUrl('getAllPrefs')).then((prefs) => setPrefs(prefs))
+    getRequest(getServerUrl('getAllPrefs')).then((prefs) => {
+      setPrefs(prefs)
+    })
   }, [])
 
-  const handleChange = ({
```

Comment:
Made a common "JsonEditor" component that can be shared by Prefs and Data View editors, so a lot of this logic is moved out of here now. :)